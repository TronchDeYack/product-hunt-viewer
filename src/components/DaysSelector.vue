<template>
  <select class="days">
    <option v-for="index in 31" :key="index" :value="index-1">
      {{ getLabel(index - 1) }}
    </option>
  </select>
</template>

<script>
export default {
  name: 'days-selector',
  methods: {
    getLabel(index) {
      if (index === 0) {
        return 'Today';
      }
      if (index === 1) {
        return 'Yesterday';
      }
      return `${index} days ago`;
    },
  },
};
</script>
