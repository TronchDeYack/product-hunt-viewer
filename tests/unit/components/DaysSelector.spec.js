import { mount } from '@vue/test-utils';

import DaysSelector from '@/components/DaysSelector.vue';

describe('DaysSelector', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(DaysSelector);
  });

  it('Should render.', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should display "Today" for the first value of the list.', () => {
    expect(wrapper.findAll('option').at(0).text()).toEqual('Today');
  });

  it('Should display "Yesterday" for the second value of the list.', () => {
    expect(wrapper.findAll('option').at(1).text()).toEqual('Yesterday');
  });

  it('Should display "2 days ago" for the third value of the list.', () => {
    expect(wrapper.findAll('option').at(2).text()).toEqual('2 days ago');
  });

  it('Should display "3 days ago" for the fourth value of the list.', () => {
    expect(wrapper.findAll('option').at(3).text()).toEqual('3 days ago');
  });
});
