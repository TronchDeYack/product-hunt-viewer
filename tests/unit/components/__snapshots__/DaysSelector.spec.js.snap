// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DaysSelector Should render. 1`] = `
<select class="days">
  <option value="0">
    Today
  </option>
  <option value="1">
    Yesterday
  </option>
  <option value="2">
    2 days ago
  </option>
  <option value="3">
    3 days ago
  </option>
  <option value="4">
    4 days ago
  </option>
  <option value="5">
    5 days ago
  </option>
  <option value="6">
    6 days ago
  </option>
  <option value="7">
    7 days ago
  </option>
  <option value="8">
    8 days ago
  </option>
  <option value="9">
    9 days ago
  </option>
  <option value="10">
    10 days ago
  </option>
  <option value="11">
    11 days ago
  </option>
  <option value="12">
    12 days ago
  </option>
  <option value="13">
    13 days ago
  </option>
  <option value="14">
    14 days ago
  </option>
  <option value="15">
    15 days ago
  </option>
  <option value="16">
    16 days ago
  </option>
  <option value="17">
    17 days ago
  </option>
  <option value="18">
    18 days ago
  </option>
  <option value="19">
    19 days ago
  </option>
  <option value="20">
    20 days ago
  </option>
  <option value="21">
    21 days ago
  </option>
  <option value="22">
    22 days ago
  </option>
  <option value="23">
    23 days ago
  </option>
  <option value="24">
    24 days ago
  </option>
  <option value="25">
    25 days ago
  </option>
  <option value="26">
    26 days ago
  </option>
  <option value="27">
    27 days ago
  </option>
  <option value="28">
    28 days ago
  </option>
  <option value="29">
    29 days ago
  </option>
  <option value="30">
    30 days ago
  </option>
</select>
`;
